bil.ott

```diff
@@ -506,13 +527,40 @@ defns reduce_exp :: '' ::=
  ----------------------------------------- :: bop_lhs
  delta |- e1 bop v2 ~> e1' bop v2
 
+ delta |- e2 ~> e2'
```

Comment:
I understand that you were following the order of the evaluation of the bop rules, but sop, cop, and bop, should be actually evaluated from left to right, missed this point in the previous commits.

Could you please, update, bop, cop, sop rules and switch to the left-to-right evaluation order?